avocado/plugins/nlist.py

```diff
@@ -38,6 +40,10 @@ def configure(self, parser):
                             action='store_true', default=False,
                             help=("Show extra information on resolution, besides "
                                   "sucessful resolutions"))
+        parser.add_argument('--write-recipes-to-directory',
+                            metavar='DIRECTORY', default=None,
+                            help=('Writes runnable recipe files to a directory'))
```

Comment:
I understood the idea, but it is not intuitive, or makes much sense, in my opinion, to have an argument that writes to files in a list command. Although it seems awkward to me, I could not think about another place to move it.

Reply:
You're right about the intuitiveness and the overall usability... I'm allowing myself to be a bit lax about it while all of this is considered experimental.

Anyway, the `nlist` command was originally conceived as `resolver`, in the sense that it would expose the behavior of the resolver code (which turns references into `ReferenceResolution`s, which can contains `Runnables`).  So, I guess we could add a "plumbing level" command such as `resolver`, which would expose those lower level features, either now, or in the future when we raise the status from experimental to "to be consumed by the general public".  I would do it later, because it looks like there's a lot that we're learning as we go, and seems a bit early to optimize on the end user experience. 

What do you think?

Reply:
I totally agree. While I find it not intuitive, I could not think about other places to add it, and the resolver command seems reasonable.

Reply:
If I wasn't clear, what I meant about "later" is: when we rename `nlist` to `list` (removing the avocado/core/loader.py based code), we will certainly have to think about a usable, compatible and sane command line interface.  And then, it may make sense to move some of those options to a "plumbling" level command or just remove them if they are found to not be useful.

Reply:
Yep, I got it!